internal/app/service/service_handler.go

```diff
@@ -38,6 +42,62 @@ func NewServiceHandlerError(msg string) *ServiceHandlerError {
 	return &ServiceHandlerError{msg: msg}
 }
 
+func (s *serviceHandler) getServiceResultsWithAggregations(filter *entity.ServiceFilter) ([]entity.ServiceResult, error) {
+	var serviceResults []entity.ServiceResult
+	servicesCiCount, err := s.database.GetServicesWithComponentInstanceCount(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	servicesImCount, err := s.database.GetServicesWithIssueMatchCount(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(servicesImCount) != len(servicesCiCount) {
+		return nil, fmt.Errorf("Error")
+	}
+
+	// don't assume that results have some order
+	// create map with id -> service
+	ciCounts := map[int64]entity.ServiceWithAggregations{}
+	imCounts := map[int64]entity.ServiceWithAggregations{}
+
+	lo.ForEach(servicesCiCount, func(s entity.ServiceWithAggregations, _ int) {
+		ciCounts[s.Id] = s
+	})
+
+	lo.ForEach(servicesImCount, func(s entity.ServiceWithAggregations, _ int) {
+		imCounts[s.Id] = s
+	})
+
+	ciIds := maps.Keys(ciCounts)
+	imIds := maps.Keys(imCounts)
+
+	slices.Sort(ciIds)
+	slices.Sort(imIds)
+
+	// check if same services were returned by the aggregation queries
+	if !slices.Equal(ciIds, imIds) {
+		return nil, fmt.Errorf("aggregation queries returned different services")
+	}
```

Comment:
Do we really want to check that? 

When a certain service has a 0 count. Is it then still be listed with count = 0 ?